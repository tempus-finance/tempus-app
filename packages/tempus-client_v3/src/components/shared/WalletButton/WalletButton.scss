$buttonHeight: 40px;
$networkSectionWidth: 55px;
$walletSectionWidth: 160px;

.tc__walletButton__disconnected {
  width: $walletSectionWidth;
  height: $buttonHeight;
  border: 1px solid var(--walletButtonBorder);
  border-radius: 4px;
  box-sizing: border-box;
  background: var(--walletButtonBackground);

  &:hover {
    background: var(--walletButtonHover);
  }
}

.tc__walletButton__connected {
  width: $networkSectionWidth + $walletSectionWidth;
  height: $buttonHeight;
  display: flex;
}

.tc__walletButton__connected-network {
  display: flex;
  align-items: center;
  justify-content: center;
  width: $networkSectionWidth;
  height: $buttonHeight;
  background: var(--walletButtonBackground);
  border: 1px solid var(--walletButtonBorder);
  box-sizing: border-box;
  border-radius: 20px 0px 0px 20px;

  &:hover {
    background: var(--walletButtonHover);
  }
}

.tc__walletButton__connected-wallet {
  display: flex;
  align-items: center;
  justify-content: center;
  width: $walletSectionWidth;
  height: $buttonHeight;
  background: var(--walletButtonBackground);
  border: 1px solid var(--walletButtonBorder);
  border-left: none;
  box-sizing: border-box;
  border-radius: 0px 20px 20px 0px;

  &:hover {
    background: var(--walletButtonHover);
  }
}

.tc__walletButton__connected-wallet-logo {
  padding-right: 8px;

  // Make react-blockies canvas round
  canvas {
    border-radius: 24px;
  }
}

.tc__walletButton__connected-wallet-info {
  width: 112px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.tc__walletButton__connected-wallet-info-balance {
  display: flex;
}
