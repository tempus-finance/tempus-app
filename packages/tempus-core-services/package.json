{
  "name": "tempus-core-services",
  "version": "0.0.1",
  "private": true,
  "description": "Tempus core Services",
  "keywords": [
    "typescript",
    "boilerplate",
    "esbuild"
  ],
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "lint": "eslint src/ --ext .js,.jsx,.ts,.tsx",
    "test": "jest",
    "clean": "rm -rf dist build package",
    "ts-node": "ts-node",
    "docs": "typedoc --entryPoints src/index.ts",
    "build": "npm run clean && tsc -p tsconfig.json",
    "build-all": "npm run clean && npm run build && npm run esbuild-node && npm run esbuild-node",
    "esbuild-node": "esbuild src/index.ts --bundle --platform=node --minify --sourcemap=external --outfile=dist/esbuild/index.js",
    "esbuild-node:dev": "esbuild src/index.ts --bundle --platform=node --sourcemap=external --outfile=dist/esbuild/index.js",
    "esbuild-node:watch": "esbuild src/index.ts --bundle --platform=node --watch --sourcemap=external --outfile=dist/esbuild/index.js"
  },
  "dependencies": {
    "@ethersproject/providers": "^5.5.2",
    "@types/jest": "^24.9.1",
    "@types/node": "^12.20.15",
    "axios": "^0.22.0",
    "date-fns": "^2.28.0",
    "ethers": "^5.4.6"
  },
  "devDependencies": {
    "esbuild": "^0.11.11",
    "eslint": "^7.22.0",
    "jest": "^26.6.3",
    "ts-jest": "^26.5.4",
    "ts-node": "^9.1.1",
    "typedoc": "^0.20.35"
  },
  "peerDependencies": {
    "@typescript-eslint/eslint-plugin": "^4.28.1",
    "@typescript-eslint/parser": "^4.28.1",
    "eslint-config-airbnb-typescript": "^12.3.1"
  }
}
