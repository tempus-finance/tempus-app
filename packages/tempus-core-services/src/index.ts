import ERC20ABI from './abi/ERC20.json';
import VaultABI from './abi/Vault.json';

export { ERC20ABI, VaultABI };

export type { ERC20 } from './abi';

export type {
  AvailableToDeposit,
  Chain,
  ChainConfig,
  Config,
  DynamicTempusPool,
  PoolShares,
  ProtocolDisplayName,
  ProtocolName,
  Ticker,
  NativeTokenTicker,
  TempusPool,
  TokenPrecision,
  TokenTypePrecision,
  YearnData,
  YearnDataApy,
  YearnDataToken,
} from './interfaces';

export type {
  DepositedEvent,
  ERC20TokenService,
  PoolBalanceChangedEvent,
  PoolBalanceChangedEventListener,
  RedeemedEvent,
  TempusAMMJoinKind,
  TempusAMMExitKind,
  TempusAMMService,
  TempusControllerService,
  TempusPoolService,
  TransferEventListener,
  VaultService,
  StatisticsService,
  StorageService,
  SwapEvent,
} from './services';

export type { Numberish } from './datastructures';

export { Decimal, ZERO, ONE, DEFAULT_DECIMAL_PRECISION } from './datastructures';

export { SwapKind } from './services';

export {
  getChainlinkFeed,
  getCoingeckoRate,
  getDefaultProvider,
  getERC20TokenService,
  initServices,
  getServices,
  getDefinedServices,
} from './services';

export {
  NumberUtils,
  DecimalUtils,
  capitalize,
  chainIdToChainName,
  chainIdHexToChainName,
  chainNameToHexChainId,
  chainToTicker,
  decreasePrecision,
  div18f,
  getCookie,
  getLibrary,
  getPastDaysNumber,
  getProvider,
  getProviderFromSignerOrProvider,
  getRangeFrom,
  getTokenPrecision,
  increasePrecision,
  isZeroString,
  mul18f,
  prettifyChainName,
  prettifyChainNameLong,
  rayToDai,
  reverseString,
  shortenAccount,
  wadToDai,
  wait,
  prettifyProtocolName,
} from './utils';

export * from './constants';
