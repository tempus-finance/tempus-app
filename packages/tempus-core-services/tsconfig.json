{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,

    "importHelpers": true,
    "experimentalDecorators": true,
    "sourceMap": true,
    "outDir": "./dist/",
    "types": ["node", "jest"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["node_modules"]
}
