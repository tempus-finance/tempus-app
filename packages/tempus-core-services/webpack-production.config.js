const path = require('path');
const webpack = require('webpack');

require('dotenv').config({ path: './.env.production' });

module.exports = {
  entry: './src/index.ts',
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        exclude: [/node_modules/],
      },
    ],
  },
  resolve: {
    extensions: ['.ts', '.js'],
  },
  output: {
    path: path.resolve(__dirname, 'dist'),
    publicPath: '',
    filename: 'index.js',
    library: 'tempus-core-services',
    libraryTarget: 'umd',
  },
  mode: 'production',
  devtool: 'eval-source-map',
  externals: ['ethers', 'rxjs', 'date-fns', '@ethersproject/providers'],
  plugins: [
    new webpack.DefinePlugin({
      'process.env': JSON.stringify(process.env),
    }),
  ],
};
