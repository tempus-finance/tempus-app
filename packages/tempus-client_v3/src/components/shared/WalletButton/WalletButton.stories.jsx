import WalletButton from './WalletButton';

export default {
  title: 'WalletButton',
  component: WalletButton,
  argTypes: {
    address: {
      control: {
        type: 'text',
      },
    },
    balance: {
      control: {
        type: 'text',
      },
    },
    ticker: {
      control: {
        type: 'text',
      },
    },
    network: {
      control: {
        type: 'select',
        options: ['ethereum', 'fantom', 'ethereum-fork'],
      },
    },
  },
};

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template = args => (
  <div style={style}>
    <WalletButton {...args} />
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  address: '0xAFE0B5E1bF4b9230A53e4A4715074ABf5B45F5de',
  balance: '34.2345',
  ticker: 'ETH',
  network: 'ethereum',
};
