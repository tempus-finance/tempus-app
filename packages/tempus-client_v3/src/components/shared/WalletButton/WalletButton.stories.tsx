import { ComponentStory, ComponentMeta } from '@storybook/react';

import WalletButton from './WalletButton';

export default {
  title: 'WalletButton',
  component: WalletButton,
  argTypes: {
    address: {
      control: {
        type: 'text',
      },
    },
    balance: {
      control: {
        type: 'text',
      },
    },
    chain: {
      control: {
        type: 'select',
        options: ['ethereum', 'fantom', 'ethereum-fork', 'unsupported'],
      },
    },
  },
} as ComponentMeta<typeof WalletButton>;

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template: ComponentStory<typeof WalletButton> = args => (
  <div style={style}>
    <WalletButton {...args} />
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  address: '0xAFE0B5E1bF4b9230A53e4A4715074ABf5B45F5de',
  balance: '34.2345',
  chain: 'ethereum',
  onConnect: () => {},
  onNetworkClick: () => {},
  onWalletClick: () => {},
};
