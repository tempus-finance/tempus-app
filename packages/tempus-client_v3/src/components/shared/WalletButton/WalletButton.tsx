import { FC, memo, useMemo } from 'react';
import Blockies from 'react-blockies';
import { Chain, chainToTicker, shortenAccount } from 'tempus-core-services';
import ButtonWrapper from '../ButtonWrapper';
import Icon from '../Icon';
import Logo from '../Logo';
import Typography from '../Typography';

import './WalletButton.scss';

interface WalletButtonProps {
  address: string;
  balance: string;
  chain: Chain;
  onConnect: () => void;
  onNetworkClick: () => void;
  onWalletClick: () => void;
}

const WalletButton: FC<WalletButtonProps> = props => {
  const { address, balance, chain, onConnect, onNetworkClick, onWalletClick } = props;

  const selectedChainTokenTicker = useMemo(() => {
    return chainToTicker(chain);
  }, [chain]);

  const shortAddress = useMemo(() => {
    return shortenAccount(address);
  }, [address]);

  const supportedChain = chain !== 'unsupported';

  return (
    <>
      {!address && (
        <ButtonWrapper className="tc__walletButton__disconnected" onClick={onConnect}>
          <Typography variant="body-primary" weight="bold">
            {/* TODO - Needs translations */}
            Connect Wallet
          </Typography>
        </ButtonWrapper>
      )}
      {address && (
        <div className="tc__walletButton__connected">
          <ButtonWrapper className="tc__walletButton__connected-network" onClick={onNetworkClick}>
            {!supportedChain && <Icon type="exclamation-error" size={20} />}
            {supportedChain && <Logo type={`token-${selectedChainTokenTicker}`} size="small" />}
          </ButtonWrapper>
          <ButtonWrapper
            className="tc__walletButton__connected-wallet"
            disabled={!supportedChain}
            onClick={onWalletClick}
          >
            {!supportedChain && (
              <Typography variant="body-primary" weight="bold">
                Unsupported
              </Typography>
            )}
            {supportedChain && (
              <>
                <div className="tc__walletButton__connected-wallet-logo">
                  <Blockies seed={address} scale={3} size={8} />
                </div>
                <div className="tc__walletButton__connected-wallet-info">
                  <Typography variant="body-tertiary" type="mono" weight="medium">
                    {shortAddress}
                  </Typography>
                  <div className="tc__walletButton__connected-wallet-info-balance">
                    <Typography variant="body-secondary" type="mono" weight="bold">
                      {balance}
                    </Typography>
                    &nbsp;
                    <Typography variant="body-secondary">{selectedChainTokenTicker}</Typography>
                  </div>
                </div>
              </>
            )}
          </ButtonWrapper>
        </div>
      )}
    </>
  );
};
export default memo(WalletButton);
