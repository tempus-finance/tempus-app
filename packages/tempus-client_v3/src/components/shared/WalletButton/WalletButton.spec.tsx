import { fireEvent, render } from '@testing-library/react';
import { chainToTicker, shortenAccount } from 'tempus-core-services';
import WalletButton, { WalletButtonProps } from './WalletButton';

jest.mock('react-blockies', () => () => (
  <canvas className="identicon" width="24" height="24" style={{ width: '24px', height: '24px' }} />
));

const mockOnConnect = jest.fn();
const mockOnNetworkClick = jest.fn();
const mockOnWalletClick = jest.fn();

const defaultProps: WalletButtonProps = {
  address: '',
  balance: '',
  chain: 'ethereum',
  onConnect: mockOnConnect,
  onNetworkClick: mockOnNetworkClick,
  onWalletClick: mockOnWalletClick,
};

const subject = (props: WalletButtonProps) => render(<WalletButton {...props} />);

describe('WalletButton', () => {
  it('renders a disconnected wallet button', () => {
    const { container, getByText } = subject(defaultProps);

    const actualWalletButton = container.querySelector('.tc__walletButton__disconnected');
    const actualText = getByText('Connect Wallet');

    expect(actualWalletButton).not.toBeNull();
    expect(actualText).not.toBeNull();

    expect(actualWalletButton).toMatchSnapshot();
  });

  it('calls `onConnect` when clicked on a disconnected wallet button', () => {
    const { queryByRole } = subject(defaultProps);

    const actualButton = queryByRole('button', { name: /Connect Wallet/ });
    expect(actualButton).not.toBeNull();

    fireEvent.click(actualButton as Element);
    expect(mockOnConnect).toHaveBeenCalledTimes(1);
  });

  it('renders a connected wallet button', () => {
    const address = '0xAFE0B5E1bF4b9230A53e4A4715074ABf5B45F5de';
    const { container, getByText } = subject({
      ...defaultProps,
      address,
      balance: '12.34',
    });

    const shortenedAccount = shortenAccount(address);

    const actualWalletButton = container.querySelector('.tc__walletButton__connected');
    const actualText = getByText(chainToTicker(defaultProps.chain));
    const actualBalance = getByText('12.34');
    const actualAddress = getByText(shortenedAccount);

    expect(actualWalletButton).not.toBeNull();
    expect(actualText).not.toBeNull();
    expect(actualBalance).not.toBeNull();
    expect(actualAddress).not.toBeNull();

    expect(actualWalletButton).toMatchSnapshot();
  });

  it('renders a connected wallet button to an unsupported network', () => {
    const address = '0xAFE0B5E1bF4b9230A53e4A4715074ABf5B45F5de';
    const { container, getByText } = subject({
      ...defaultProps,
      address,
      balance: '77.99',
      chain: 'unsupported',
    });

    const actualWalletButton = container.querySelector('.tc__walletButton__connected');
    const actualText = getByText('Unsupported');

    expect(actualWalletButton).not.toBeNull();
    expect(actualText).not.toBeNull();

    expect(actualWalletButton).toMatchSnapshot();
  });
});
