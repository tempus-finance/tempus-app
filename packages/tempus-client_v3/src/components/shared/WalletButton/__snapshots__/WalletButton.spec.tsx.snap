// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletButton renders a connected wallet button 1`] = `
<div
  class="tc__walletButton__connected"
>
  <button
    class="tc__btn tc__walletButton__connected-network"
    data-selected="false"
    type="button"
  >
    <svg
      class="tc__logo tc__logo-token-ETH"
      fill="none"
      height="24"
      viewBox="0 0 40 40"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 40C31.0457 40 40 31.0457 40 20C40 8.9543 31.0457 0 20 0C8.9543 0 0 8.9543 0 20C0 31.0457 8.9543 40 20 40Z"
        fill="#627EEA"
      />
      <path
        clip-rule="evenodd"
        d="M11.25 20.275L20.6225 5V16.0875L11.25 20.275ZM20.6225 27.4588V34.9938L11.25 22.0201L20.6225 27.4588Z"
        fill="white"
        fill-rule="evenodd"
      />
      <path
        d="M20.625 25.7163L29.9962 20.275L20.625 16.09V25.7163Z"
        fill="white"
        fill-opacity="0.2"
      />
      <path
        clip-rule="evenodd"
        d="M20.625 5V16.0875L29.9962 20.275L20.625 5ZM20.625 27.4573V34.991L30.0025 22.0173L20.625 27.4573ZM20.6225 25.7163L11.25 20.275L20.6225 16.09V25.7163Z"
        fill="white"
        fill-opacity="0.6"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="tc__btn tc__walletButton__connected-wallet"
    data-selected="false"
    type="button"
  >
    <div
      class="tc__walletButton__connected-wallet-logo"
    >
      <canvas
        class="identicon"
        height="24"
        style="width: 24px; height: 24px;"
        width="24"
      />
    </div>
    <div
      class="tc__walletButton__connected-wallet-info"
    >
      <div
        style="font-style: normal; font-size: 10px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
      >
        0xAFE0...5F5de
      </div>
      <div
        class="tc__walletButton__connected-wallet-info-balance"
      >
        <div
          style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'Azeret Mono', monospace; opacity: 1;"
        >
          12.34
        </div>
        Â 
        <div
          style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          ETH
        </div>
      </div>
    </div>
  </button>
</div>
`;

exports[`WalletButton renders a connected wallet button to an unsupported network 1`] = `
<div
  class="tc__walletButton__connected"
>
  <button
    class="tc__btn tc__walletButton__connected-network"
    data-selected="false"
    type="button"
  >
    <svg
      class="tc__icon tc__icon-exclamation-error"
      fill="none"
      height="20"
      viewBox="0 0 16 16"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.04566 1.54327C7.46762 0.81891 8.53238 0.81891 8.95434 1.54327L15.8536 13.3868C16.2723 14.1056 15.7432 15 14.8992 15H1.10078C0.256824 15 -0.272275 14.1056 0.146446 13.3868L7.04566 1.54327Z"
        fill="#FF0F0F"
      />
      <path
        d="M7 6.00314V8.98597C7 9.54015 7.44767 9.9894 7.9999 9.9894C8.55221 9.9894 8.99991 9.54003 8.9998 8.98577L8.9992 6.00293C8.99909 5.45985 8.56833 5.01548 8.02737 5.0004C7.46465 4.9847 7 5.43821 7 6.00314Z"
        fill="white"
      />
      <path
        d="M8.9995 11.997C8.99978 11.4427 8.55209 10.9929 7.99975 10.9929C7.4476 10.9929 7 11.4424 7 11.9965C7 12.5506 7.4476 13 7.99975 13C8.5517 13 8.99922 12.5509 8.9995 11.997Z"
        fill="white"
      />
    </svg>
  </button>
  <button
    class="tc__btn tc__walletButton__connected-wallet"
    data-selected="false"
    disabled=""
    type="button"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Unsupported
    </div>
  </button>
</div>
`;

exports[`WalletButton renders a disconnected wallet button 1`] = `
<button
  class="tc__btn tc__walletButton__disconnected"
  data-selected="false"
  type="button"
>
  <div
    style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    Connect Wallet
  </div>
</button>
`;
